### ISSUE TITLE
Leftover height when narrowing candidates

### ISSUE THREAD

**Issue by elken** [minad/corfu#186]
![image](https://user-images.githubusercontent.com/2872862/172104412-2ccce536-1504-4cd8-9185-5c32ac35eaec.png)

Screenshot above describes the issue; when initiating completion, the frame is populated fine. However, once I type a few more characters to narrow the selection, the "old" height still persists. Maybe #149 as I make `'corfu-default` inherit from `'tooltip`?. There are only 2 valid candidates, so I would expect the height to reflect that.

I'm not sure if this is a bug or a feature, if it's the former how simple would it be to add a toggle for that behaviour?

[config](https://github.com/elken/doom/tree/master/modules/completion/corfu)

**Comment by minad** [comment]
Ouch! These child frame resizing issues should be sorted out by now. Can you reproduce this in emacs -Q? Which emacs version is this? Pgtk? Which Corfu commit?

**Comment by minad** [comment]
I think it is this ugly bug: https://github.com/minad/corfu/issues/17

**Comment by minad** [comment]
Please check if the issue occurs before: https://github.com/minad/corfu/commit/a262a214325c7e33704bb1fade0fa40042ff93a5

This could be a regression due to @galeo's recent corfu-doc related refactorings

**Comment by elken** [comment]
> Can you reproduce this in emacs -Q? 

I went to try it on macos and it works as expected here, which "helps".

> Which emacs version is this? 

28.1

> Which GUI toolkit? 

X11

> Which Corfu commit?

corfu is unpinned so whatever the latest commit is

> Please check if the issue occurs before: https://github.com/minad/corfu/commit/a262a214325c7e33704bb1fade0fa40042ff93a5

I will do this after work tonight along with `emacs -Q` and see what comes back :)

**Comment by minad** [comment]
Please check the newest commit. This is probably just a regression.

**Comment by elken** [comment]
> Please check the newest commit. This is probably just a regression. (should work on both emacs -Q and in Doom)

I will report back one way or another after testing, thanks as always minad :)

**Comment by minad** [comment]
One more question - do you use gtk3/gnome/cinnamon? The aforementioned bug is specific to those gui toolkits.

**Comment by elken** [comment]
> One more question - do you use gtk3 and (gnome or cinnamon)? The aforementioned bug is specific to those gui toolkits.

I do :) I edited my original comment a few times since posting it

**Comment by elken** [comment]
This is indeed still an issue I'm afraid :( 
![image](https://user-images.githubusercontent.com/2872862/172241336-082ae1df-2f74-4bbd-8787-ff4ad69ac408.png)

**Comment by minad** [comment]
Can you test the latest stable version?

**Comment by elken** [comment]
The elpa build from this morning? Also fails

**Comment by elken** [comment]
I'll reset to the commit hash from earlier

**Comment by minad** [comment]
Please check this commit: https://github.com/minad/corfu/commit/a262a214325c7e33704bb1fade0fa40042ff93a5

And also the latest stable version!

This would be much appreciated!

**Comment by elken** [comment]
Yup works as intended on e875b467e2a196ed7e470a593e14bcac4399aeb2

**Comment by minad** [comment]
Thanks! I reverted back to a working version. Please confirm!

Btw, if you create a doom module I can help out with some review. Is there a PR to look at?

**Comment by elken** [comment]
> Thanks! I reverted back to a working version. Please confirm!

Will do shortly!

> Btw, if you create a doom module I can help out with some review. Is there a PR to look at?

That would be GREATLY appreciated, but atm I'm still ironing out the best all-bases setup. There is another issue I'll raise tomorrow on snippet expansion but I'm about to turn in soon :) 

I'll certainly ping you when it gets close to review

**Comment by elken** [comment]
This seems to be working now! Thanks minad :) I consider this properly resolved now

**Comment by minad** [comment]
@elken Just fyi, there seems to be still an lsp-mode-related issue #188, https://github.com/emacs-lsp/lsp-mode/issues/3555. This is something which should be expected. Since for many years Company was the only option (except maybe autocomplete a bit longer ago), most packages were developed against Company and only tested against Company. This means that there could be subtle compatibility issues. I hope that these can be sorted out step by step. Furthermore since Company was the only option, many packages come with Company-specific backends or there exist many of those company-* packages, which could in principle be implemented without a Company dependency (see what I wrote in https://github.com/minad/cape/#company-adapter) or even just as a Capf. I am not yet 100% sure if this is already realized, but `cape-company-to-capf` at least aims to be fully compatible with Company, so in the long term there shouldn't be a problem to handle all those backends.

**Comment by galeo** [comment]
Hi @minad 

Maybe the value of `x-gtk-resize-child-frames` should also be set before `set-frame-size`. I can't make a test though. I see you reverted the commit to a262a214325c7e33704bb1fade0fa40042ff93a5. However, the right way to solve the problem should be found.

@elken Can you help test the code in #189 to see if it works?

Thanks.

**Comment by elken** [comment]
Looking now :)
